### PR TITLE
[fix](planner) let OlapScanNode turn off preaggragation when there is a filter on DELETE_SIGN

### PR DESCRIPTION
**Pull request by dataroaring**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
[This issue](https://github.com/apache/doris/issues/12089) seems related to this PR

File: be/src/exec/olap_scanner.cpp [hunk 1]
```diff
@@ -291,20 +292,23 @@ Status OlapScanner::_init_return_columns(bool need_seq_col) {
         _query_slots.push_back(slot);
     }
 
-    // expand the sequence column
-    if (_tablet_schema->has_sequence_col() && need_seq_col) {
-        bool has_replace_col = false;
+    has_replace_col = false;
```
Comment (HappenLee) [inline be/src/exec/olap_scanner.cpp:295]: the value seems always `false` ？

File: regression-test/suites/delete_p0/test_delete_sign.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+DROP TABLE IF EXISTS delete_sign_test;
+
+CREATE TABLE delete_sign_test (
+                `uid` BIGINT NULL,
+                `v1` BIGINT NULL 
+                )
+UNIQUE KEY(uid)
+DISTRIBUTED BY HASH(uid) BUCKETS 3
+PROPERTIES (
```
Comment (zhannngchen) [inline regression-test/suites/delete_p0/test_delete_sign.sql:9]: Could you help to add a case for MoW?

**Review by zhannngchen — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 3]
```diff
@@ -1044,6 +1045,7 @@ private void filterDeletedRows(Analyzer analyzer) throws AnalysisException {
             Expr conjunct = new BinaryPredicate(BinaryPredicate.Operator.EQ, deleteSignSlot, new IntLiteral(0));
             conjunct.analyze(analyzer);
             conjuncts.add(conjunct);
+            closePreAggregation(Column.DELETE_SIGN + " is used as conjuncts.");
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:1048]: MoW requires isPreAggregation on, should use `setIsPreAggregation(false, reason)`?

**Review by morrySnow — APPROVED** [review]

**Review by zhannngchen — COMMENTED** [review]
LGTM

**Pull request merged**